matchbox-server/src/test/java/ch/ahdis/matchbox/test/MatchboxApiR5Test.java

```diff
@@ -0,0 +1,226 @@
+      IBaseOperationOutcome theOutcome) {
+    if (theOutcome == null) {
+      return null;
+    }
+    RuntimeResourceDefinition ooDef = theCtx.getResourceDefinition(theOutcome);
+    BaseRuntimeChildDefinition issueChild = ooDef.getChildByName("issue");
+    List<IBase> issues = issueChild.getAccessor().getValues(theOutcome);
+    if (issues.isEmpty()) {
+      return null;
+    }
+    IBase issue = issues.get(0);
+    if (issue instanceof IBaseHasExtensions) {
+      List<? extends IBaseExtension<?, ?>> extensions = ((IBaseHasExtensions) issue).getExtension();
+      for (IBaseExtension<?, ?> nextSource : extensions) {
+        if (nextSource.getUrl().equals("http://matchbox.health/validiation")) {
+          return nextSource;
+        }
+      }
+    }
+    return null;
+  }
+
+  public String getSessionId(FhirContext ctx, IBaseOperationOutcome outcome) {
+    IBaseExtension<?, ?> ext = getMatchboxValidationExtension(ctx, outcome);
+    List<IBaseExtension<?, ?>> extensions = (List<IBaseExtension<?, ?>>) ext.getExtension();
+    for (IBaseExtension<?, ?> next : extensions) {
+      if (next.getUrl().equals("sessionId")) {
+        IPrimitiveType<?> value = (IPrimitiveType<?>) next.getValue();
+        return value.getValueAsString();
+      }
+    }
+    return null;
+  }
+
+  public String getIg(FhirContext ctx, IBaseOperationOutcome outcome) {
+    IBaseExtension<?, ?> ext = getMatchboxValidationExtension(ctx, outcome);
+    List<IBaseExtension<?, ?>> extensions = (List<IBaseExtension<?, ?>>) ext.getExtension();
+    for (IBaseExtension<?, ?> next : extensions) {
+      if (next.getUrl().equals("ig")) {
+        IPrimitiveType<?> value = (IPrimitiveType<?>) next.getValue();
+        return value.getValueAsString();
+      }
+    }
+    return null;
+  }
+
+  public String getTxServer(FhirContext ctx, IBaseOperationOutcome outcome) {
+    IBaseExtension<?, ?> ext = getMatchboxValidationExtension(ctx, outcome);
+    List<IBaseExtension<?, ?>> extensions = (List<IBaseExtension<?, ?>>) ext.getExtension();
+    for (IBaseExtension<?, ?> next : extensions) {
+      if (next.getUrl().equals("txServer")) {
+        IPrimitiveType<?> value = (IPrimitiveType<?>) next.getValue();
+        return value.getValueAsString();
+      }
+    }
+    return null;
+  }
+
+  @Test
+  public void validatePatientRawR5() {
+    FhirContext contextR4 = FhirVersionEnum.R4.newContext();
+    ValidationClient validationClient = new ValidationClient(contextR4, this.targetServer);
+
+    String patient = "<Patient xmlns=\"http://hl7.org/fhir\">\n" + "            <id value=\"example\"/>\n"
+        + "            <text>\n" + "               <status value=\"generated\"/>\n"
+        + "               <div xmlns=\"http://www.w3.org/1999/xhtml\">42 </div>\n" + "            </text>\n"
+        + "         </Patient>\n";
+
+    IBaseOperationOutcome operationOutcome = validationClient.validate(patient,
+        "http://hl7.org/fhir/StructureDefinition/Patient|5.0.0");
```

Comment:
the |5.0.0 should not be necessary when in this test it is "only" a R5 test instance?

Reply:
It still loads the FHIR Core R4 IG because that's the default configuration, it's not overridden by the test-r5 profile. Not sure how to best override it.